File: api/vehicle-service/src/main/java/com/carrental/vehicleservice/model/dto/VehicleOptionsWithAssocCountDTO.java
package com.carrental.vehicleservice.model.dto;

import lombok.Data;

import java.util.Set;

@Data
public class VehicleOptionsWithAssocCountDTO {

    private Set<AssocDetailsDTO> bodyTypes;

    private Set<AssocDetailsDTO> brands;

    private Set<ModelAssocDetailsDTO> models;

    private Set<AssocDetailsDTO> colors;

    private Set<AssocDetailsDTO> fuelTypes;
}

File: frontend-app/shared/model/VehicleOptionsWithAssocCountDTO.ts
import { AssocDetailsDTO } from './AssocDetailsDTO';
import { ModelAssocDetailsDTO } from './ModelAssocDetailsDTO';

export class VehicleOptionsWithAssocCountDTO {
    bodyTypes: AssocDetailsDTO[];
    brands: AssocDetailsDTO[];
    models: ModelAssocDetailsDTO[];
    colors: AssocDetailsDTO[];
    fuelTypes: AssocDetailsDTO[];

    constructor(
        bodyTypes: AssocDetailsDTO[],
        brands: AssocDetailsDTO[],
        models: ModelAssocDetailsDTO[],
        colors: AssocDetailsDTO[],
        fuelTypes: AssocDetailsDTO[]
    ) {
        this.bodyTypes = bodyTypes;
        this.brands = brands;
        this.models = models;
        this.colors = colors;
        this.fuelTypes = fuelTypes;
    }
}
